tensorflow_io/avro/20190307-avro-dataset.md

```diff
@@ -0,0 +1,299 @@
+         "type":"array",
+         "items":{  
+            "name":"coworkers",
+            "type":"array",
+            "items":"com.test.person"
+         }
+      },
+      {  
+         "name":"car",
+         "type":"record",
+         "fields":[  
+            {  
+               "name":"color",
+               "type":"string"
+            },
+            {  
+               "name":"engine",
+               "type":"record",
+               "fields":[  
+                  {  
+                     "name":"id",
+                     "type":"int"
+                  },
+                  {  
+                     "name":"power",
+                     "type":"float"
+                  },
+                  {  
+                     "name":"cylinders",
+                     "type":"int"
+                  },
+                  {  
+                     "name":"sparkplugs",
+                     "type":"int"
+                  }
+               ]
+            }
+         ]
+      },
+      {  
+         "name":"cars",
+         "type":"map",
+         "values":"com.test.car"
+      }
+   ]
+}
+```
+**Listing 1** defines a person with name, address, friends, jobs, and cars.  Note, that the definition of the nested person would require a union with a nested type to be valid--but to simplify we left this out here.
+
+We start of the discussion of the API through examples (see Table 2 and Table 3). These examples list the key expressions together with an explanation. The status indicates whether this is a new, or the same as in the existing API and if such a change would be backward incompatible.
+
+| **Key Expressions** | **Brief Explanation** |
+| --- | --- |
+| name.first | Get the first name of the name record. |
+| friends[2].name.first | Get the 3rd friend&#39;s first name. |
+| friends[\*].name.first | Get all friend&#39;s first names. |
+| friends[\*].address[\*].street | Get all friend&#39;s addresses street name. In this example we have two dimensions. |
+| friends[\*].jobs[\*].coworkers[\*].name.first | Get the first names of all coworkers from all jobs that your friends have had. In this example we have three dimensions. Each \* adds a dimension. |
+| cars[&#39;nickname&#39;].color | Get the color of your car with that nickname from a map. |
+| friends[gender=&#39;unknown&#39;].name.first | Get the first name of all friends whose gender is unknown. |
+| friends[name.first=name.last].name.initial | Get initials from all the friends where their first name matches their last name. |
+| friends[name.first=@name.first].name.initial | Get initials from all friends that have the same first name as the person itself. Need some identifier to indicate that we would like to start searching from the root--here @. |
+
+**Table 2** describes the key expressions and lists their status. New means that this was not supported before. Same means it already exists. Break means that this would introduce a backward incompatible change.
```

Comment:
Should we remove the `lists their status` part and new/same distinction? I don't see a status column or any expressions marked as "new".